algebird-core/src/main/scala/com/twitter/algebird/macros/GroupMacro.scala

```diff
@@ -0,0 +1,45 @@
+package com.twitter.algebird.macros
+
+import scala.language.experimental.{ macros => sMacros }
+import scala.reflect.macros.Context
+import scala.reflect.runtime.universe._
+
+import com.twitter.algebird._
+
+object GroupMacro {
+  def caseClassGroup[T](c: Context)(implicit T: c.WeakTypeTag[T]): c.Expr[Group[T]] = {
+    import c.universe._
+
+    macros.ensureCaseClass(c)
+
+    val res = q"""
+    new _root_.com.twitter.algebird.Group[$T] {
+      ${SemigroupMacro.plus(c)}
+      ${SemigroupMacro.sumOption(c)}
+      ${MonoidMacro.zero(c)}
+      ${negate(c)}
+      ${minus(c)}
+    }
+    """
+    c.Expr[Group[T]](res)
+  }
+
+  def negate[T](c: Context)(implicit T: c.WeakTypeTag[T]): c.Tree = {
+    import c.universe._
+
+    val companion = macros.getCompanionObject(c)
+    val negateList = getParams(c).map(param => q"implicitly[_root_.com.twitter.algebird.Group[$param]].negate(x.$param)")
```

Comment:
how is this working `Group[$param]`? Isn't `param` the name of the accessor, not the type?